packages/gatsby-plugin-image/src/transforms/gatsby-plugin-image.js

```diff
@@ -0,0 +1,340 @@
+            path.node.object,
+            t.identifier(`gatsbyImageData`)
+          )
+          path.replaceWith(updatedExpression)
+          state.opts.hasChanged = true
+        }
+      },
+      TaggedTemplateExpression({ node }, state) {
+        if (node.tag.name !== `graphql`) {
+          return
+        }
+        const query = node.quasi?.quasis?.[0]?.value?.raw
+        if (query) {
+          const {
+            ast: transformedGraphQLQuery,
+            hasChanged,
+          } = processGraphQLQuery(query)
+
+          if (hasChanged) {
+            node.quasi.quasis[0].value.raw = graphql.print(
+              transformedGraphQLQuery
+            )
+            state.opts.hasChanged = true
+          }
+        }
+      },
+      CallExpression({ node }, state) {
+        if (node.callee.name !== `graphql`) {
+          return
+        }
+        const query = node.arguments?.[0].quasis?.[0]?.value?.raw
+
+        if (query) {
+          const {
+            ast: transformedGraphQLQuery,
+            hasChanged,
+          } = processGraphQLQuery(query)
+
+          if (hasChanged) {
+            node.arguments[0].quasis[0].value.raw = graphql.print(
+              transformedGraphQLQuery
+            )
+            state.opts.hasChanged = true
+          }
+        }
+      },
+    },
+  }
+}
+
+function processImportUsage(path, t, template, state) {
+  const node = path.parent
+
+  if (!t.isJSXOpeningElement(node)) {
+    console.log(
+      `It appears you're using the image component for something this codemod does not currently support. You will need to change the reference in ${state.opts.filename} manually.`
+    )
+    return
+  }
+
+  const componentName = t.jsxIdentifier(`GatsbyImage`)
+
+  const fixedOrFluid = node.attributes.filter(({ name }) =>
+    propNames.includes(name?.name)
+  )
+
+  const otherAttributes = node.attributes.filter(
+    ({ name }) => !propNames.includes(name?.name)
+  )
+
+  if (!fixedOrFluid.length > 0) {
+    path.parentPath.replaceWith(
+      t.jsxOpeningElement(componentName, [...otherAttributes], true)
+    )
+    return
+  }
+  const expressionValue = fixedOrFluid?.[0]?.value?.expression
+
+  let newImageExpression = expressionValue // by default, pass what they pass
+  if (
+    t.isMemberExpression(expressionValue) &&
+    propNames.includes(expressionValue?.property.name)
+  ) {
+    if (expressionValue?.object?.object) {
+      newImageExpression = template.expression
+        .ast`${expressionValue?.object?.object}.childImageSharp.gatsbyImageData`
+    } else if (expressionValue?.object) {
+      newImageExpression = template.expression
+        .ast`${expressionValue?.object}.gatsbyImageData`
+    }
+
+    newImageExpression.extra.parenthesized = false // the template adds parens and we don't want it to
+  } else if (t.isObjectExpression(expressionValue)) {
+    expressionValue.properties.map(item => {
+      if (item.key.name !== `src`) return
```

Comment:
I had one error where adding a "?" here as well resolved things. 

```suggestion
      if (item.key?.name !== `src`) return
```

Reply:
Oh right, `key` doesn't exist if `item` is a `SpreadElement`.

Reply:
Thanks for the reminder! Forgot about this one.